src/erlcloud_aws.erl

```diff
@@ -574,6 +590,51 @@ prop_to_list_defined( Name, Props ) ->
     end.
 
 
+-spec get_role_credentials(aws_config()) -> {ok, #role_credentials{}} | {error, term()}.
+get_role_credentials(#aws_config{assume_role = AssumeRoleOptions} = Config) ->
+    RoleArn = proplists:get_value(role_arn, AssumeRoleOptions, undefined),
+    case application:get_env(erlcloud, {role_credentials, RoleArn}) of
+        {ok, #role_credentials{expiration_gregorian_seconds = Expiration} = Credentials} ->
+            Now = calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
+            %% Get new credentials if these will expire in less than 2 minutes
+            case Expiration - Now < 120 of
```

Comment:
I will change to 300 secs again but i do not see any advantage of put this a consumer config options.